projects/admin/src/app/record/detail-view/document-detail-view/document-detail-view.component.ts

```diff
@@ -202,4 +204,61 @@ export class DocumentDetailViewComponent implements DetailRecord, OnInit, OnDest
     }
     return 'standard';
   }
+
+  /**
+   * Format "part of" numbering for display
+   *
+   * @param num: numbering to format
+   * @return formatted numbering (example: 2020, vol. 2, nr. 3, p. 302)
+   */
+  formatNumbering(num: any) {
+    const numbering = [];
+    if (num.year) {
+      numbering.push(num.year);
+    }
+    if (num.volume) {
+      const volume = [this._translateService.instant('vol'), num.volume];
+      numbering.push(volume.join('. '));
+    }
+    if (num.issue) {
+      const issue = [this._translateService.instant('nr'), num.issue];
+      numbering.push(issue.join('. '));
+    }
+    if (num.pages) {
+      const pages = [this._translateService.instant('p'), num.pages];
+      numbering.push(pages.join('. '));
+    }
+    return numbering.join(', ');
+  }
+
+  /**
+   * Get "part of" label from host document type
+   * @param hostDocument - host document
+   * @return corresponding translated label
+   */
+  getPartOfLabel(hostDocument: any) {
+    if (hostDocument.metadata.issuance.subtype === 'periodical') {
+      return this._translateService.instant('Journal');
+    }
+    if (hostDocument.metadata.issuance.subtype === 'monographicSeries') {
+      return this._translateService.instant('Series');
+    }
+    return this._translateService.instant('Published in');
+  }
+
+  /**
+   * Get short main title
+   * @param titles - document titles
+   * @return - main title to display
+   */
+  getShortMainTitle(titles: any) {
+    const bfTitles: Array<any> = titles.filter((title: any) => title.type === 'bf:Title');
+    for (const bfTitle of bfTitles) {
+      for (const mainTitle of bfTitle.mainTitle) {
+        if (!mainTitle.language) {
+          return mainTitle.value;
+        }
+      }
+    }
+  }
```

Comment:
Maybe it will great to transfer this code into a new Pipe because it will be probably used into other component in the future. Is it a little bit strange to define a short title as a mainTitle without language ?!?

Reply:
Actually, I was supposed to display a short title (ui_title_text). But it's not reachable with the pipe or the method `getRecord`. So Johnny suggested to put this one instead. There is already a `mainTitle` pipe, to display all the main titles. I think this case is more particular, but I can move it in a pipe. Actually, in the existing pipe, with a parameter. What do you think?

Reply:
It's not mandatory at this time so don't touch the code and go forward ;-)